tests/on_target/tests/conftest.py

```diff
@@ -60,11 +60,23 @@ def t91x_board():
         pytest.fail("No UARTs found")
     log_uart_string = all_uarts[0]
     uart = Uart(log_uart_string, timeout=UART_TIMEOUT)
+
+    yield types.SimpleNamespace(
+		uart=uart
+		)
+
+    uart_log = uart.whole_log
+    uart.stop()
+    recover_device()
+
+    scan_log_for_assertions(uart_log)
+
+@pytest.fixture(scope="module")
+def t91x_fota():
```

Comment:
we could also make t91x_fota inherit uart from t91x_board, like this:
`@pytest.fixture(scope="module")
def t91x_fota(t91x_board):
    # Use t91x_board fixture to provide UART functionality to t91x_fota
    uart = t91x_board.uart
`

Reply:
I think this will be cleanest. One would not use the fota fixture without the uart fixture.